File: frontend/src/api/stats.ts
import request from './request'

export interface Stats {
  quotes: number
  images: number
  published: number
}

export interface Activity {
  id: number
  type: string
  content: string
  created_at: string
}

// 获取统计数据
export const getStats = () => {
  return request.get<Stats>('/stats')
}

// 获取最近活动
export const getRecentActivities = (limit: number = 10) => {
  return request.get<Activity[]>('/stats/activities', { params: { limit } })
} 

File: backend/app/routers/stats.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.database import get_db
from app.models.base import Quote, Image, Activity
from app.schemas.stats import Stats, Activity as ActivitySchema
from typing import List

router = APIRouter()

@router.get("/", response_model=Stats)
async def get_stats(db: Session = Depends(get_db)):
    """获取统计数据"""
    quotes_count = db.query(Quote).count()
    images_count = db.query(Image).count()
    published_count = db.query(Activity).filter(Activity.type == "publish").count()
    
    return {
        "quotes": quotes_count,
        "images": images_count,
        "published": published_count
    }

@router.get("/activities", response_model=List[ActivitySchema])
async def get_activities(
    limit: int = 10,
    db: Session = Depends(get_db)
):
    """获取最近活动"""
    activities = db.query(Activity)\
        .order_by(Activity.created_at.desc())\
        .limit(limit)\
        .all()
    return activities 